### DISCUSSION TITLE
Libraries using Wire.h do not compile on ATTiny85

### DISCUSSION THREAD

**Discussion by ParamveerGSingh** [SpenceKonde/ATTinyCore#584] [Library discussion]
Hi.. I am trying to upload a program on ATTiny that uses AdafruitBNO055 library and it does not compile and throws some errors. can you help me out with this.. Thank you.

PFA the code used and the error.

First tried it with TinyWirem then with Wire.h both give the same error
[code and error.pdf](https://github.com/SpenceKonde/ATTinyCore/files/6656376/code.and.error.pdf)

**Comment by SpenceKonde** [comment]
You're using the stock version 9f wire.h, this cannot happen if you are
using attinycore board definition,  (and will happen if you try to use
those libraries on any core for the attiny85 other than ATTinyCore. All
those errors about TW__ not declared are because stock version of wire is
being used on part that it doesnt support. A huge amount of effort a few
years back went I to a shapesbigti g wire library that picks one of three
completely separate implementations based on the part. So I think even if
you have my core installed, you also have some other core for t85 installed
and are using that one

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by SpenceKonde** [comment]
Just read over that again. I think I may see your problem. You are
using the
Windows Store version of the Arduino IDE. It suffers frp, ALL MANNER of
wacky problems. Our support for ATTinyCore/Windows Store IDE
specific issues is limited to merging pull requests or making changes which
are described in detail by those who have the necessary experience and
skill to provide such help[
I do not consider it fit for purpose due to the variety of bugs it
manifests (which are not consistent for all users on apparently
identical machines!)\

Frankly. I am aware of not a single piece of good software in the windows
app store. The stuff that was designed for that from the start;lacks
features,and is terribly dumbed down  (unsuited to anyone smart enough to
program embedded systems) while the desktop apps ported to windows store to
are plagued by weirdo bugs that are generally impossible to resolve.

On Tue, Jun 15, 2021 at 10:32 AM ParamveerGSingh ***@***.***>
wrote:

> Hi.. I am trying to upload a program on ATTiny that uses AdafruitBNO055
> library and it does not compile and throws some errors. can you help me out
> with this.. Thank you.
>
> PFA the code used and the error.
>
> First tried it with TinyWirem then with Wire.h both give the same error
> code and error.pdf
> <https://github.com/SpenceKonde/ATTinyCore/files/6656376/code.and.error.pdf>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/ATTinyCore/discussions/584>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW7USNSMTGA6TBIEI33TS5P75ANCNFSM46XM356Q>
> .
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Reply by ParamveerGSingh** [reply]
Indeed.. I am using the windows store version.. Thanks for the help. I deleted it and got a different version. and now it works
